[Mainnet Bug] Sending too little of native token with gas dropoff results in an error

## Describe the issue in detail

 - https://mainnet--wormhole-connect.netlify.app/ as of 5/3 10AM (v0.0.1 on NPM)
 - Mainnet Avalanche -> Sui with gas drop-off. Select a small amount of AVAX (e.g. 0.01), slide gas dropoff all the way to the right until you see that you're sending 0 AVAX
 - Attempt to make a transaction, see that it fails:
![screenshot](https://i.imgur.com/YdL32SM.png)

```
main.5ded1fe2.js:137 Error: cannot estimate gas; transaction may fail or may require manual gas limit [ See: https://links.ethers.org/v5-errors-UNPREDICTABLE_GAS_LIMIT ] (reason="execution reverted: insufficient amount", method="estimateGas", transaction={"from":"0x49887A216375FDED17DC1aAAD4920c3777265614","to":"0x461a8878060Aa45Fa685732bfF654Ca1FDeF2830","value":{"type":"BigNumber","hex":"0x2386f26fc10000"},"data":"0x29ac8361000000000000000000000000000000000000000000000000002386f26fc100000000000000000000000000000000000000000000000000000000000000000015d5c67d73166147f6fec91717187651966cc15c5caec2462dbbe380f44b21e87f0000000000000000000000000000000000000000000000000000000000000000","accessList":null}, error={"code":-32603,"message":"Internal JSON-RPC error.","data":{"code":3,"message":"execution reverted: insufficient amount","data":"0x08c379a000000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000013696e73756666696369656e7420616d6f756e7400000000000000000000000000"},"stack":"{\n  \"code\": -32603,\n  \"message\": \"Internal JSON-RPC error.\",\n  \"data\": {\n    \"code\": 3,\n    \"message\": \"execution reverted: insufficient amount\",\n    \"data\": \"0x08c379a000000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000013696e73756666696369656e7420616d6f756e7400000000000000000000000000\"\n  },\n  \"stack\": \"Error: Internal JSON-RPC error.\\n    at new i (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-1.js:21:14454)\\n    at s (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-1.js:21:17394)\\n    at Object.internal (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-1.js:21:18004)\\n    at c (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/background-0.js:1:129727)\\n    at chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/background-0.js:1:130726\\n    at async chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/background-3.js:10:64901\"\n}\n  at new i (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-1.js:21:14454)\n  at s (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-1.js:21:17394)\n  at Object.internal (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-1.js:21:18004)\n  at c (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/background-0.js:1:129727)\n  at chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/background-0.js:1:130726\n  at async chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/background-3.js:10:64901"}, code=UNPREDICTABLE_GAS_LIMIT, version=providers/5.7.2)
    at l.makeError (main.5ded1fe2.js:37:6516643)
    at l.throwError (main.5ded1fe2.js:37:6516763)
    at Cp (main.5ded1fe2.js:37:9488176)
    at Fp.<anonymous> (main.5ded1fe2.js:37:9498226)
    at Generator.throw (<anonymous>)
    at s (main.5ded1fe2.js:37:9487264)
    ```

## What is the expected/desired behavior

- Transaction should not fail - gas drop-off slider should not allow you to send 0 of the native token

--- a/wormhole-connect/src/sdk/index.ts
+++ b/wormhole-connect/src/sdk/index.ts
@@ -100,7 +100,6 @@ export const parseMessageFromTx = async (
   };
   const decimals = await fetchTokenDecimals(tokenId, parsed.fromChain);
   const token = getTokenById(tokenId);
-  console.log(token, tokenId);
 
   const base: ParsedMessage = {
     ...parsed,

--- a/wormhole-connect/src/views/Bridge/NativeGasSlider.tsx
+++ b/wormhole-connect/src/views/Bridge/NativeGasSlider.tsx
@@ -100,9 +100,8 @@ const INITIAL_STATE = {
 function GasSlider(props: { disabled: boolean }) {
   const { classes } = useStyles();
   const dispatch = useDispatch();
-  const { token, toNetwork, amount, maxSwapAmt, destGasPayment } = useSelector(
-    (state: RootState) => state.transfer,
-  );
+  const { token, toNetwork, amount, maxSwapAmt, destGasPayment, relayerFee } =
+    useSelector((state: RootState) => state.transfer);
   const { receiving: receivingWallet } = useSelector(
     (state: RootState) => state.wallet,
   );
@@ -117,15 +116,19 @@ function GasSlider(props: { disabled: boolean }) {
   useEffect(() => {
     if (!amount || !maxSwapAmt || destGasPayment === PaymentOption.MANUAL)
       return;
+    // multiply by 0.995 to avoid errors due to rounding, low amount, low gas, etc.
+    const amountWithoutRelayerFee = (amount - (relayerFee || 0)) * 0.995;
     const actualMaxSwap =
-      amount && maxSwapAmt && maxSwapAmt > amount ? amount : maxSwapAmt;
+      amount &&
+      maxSwapAmt &&
+      Math.max(Math.min(maxSwapAmt, amountWithoutRelayerFee), 0);
     const newTokenAmount = toFixedDecimals(`${amount - state.swapAmt}`, 6);
     setState((prevState) => ({
       ...prevState,
       token: Number.parseFloat(newTokenAmount),
       max: actualMaxSwap,
     }));
-  }, [maxSwapAmt, amount, destGasPayment, state.swapAmt]);
+  }, [maxSwapAmt, amount, destGasPayment, state.swapAmt, relayerFee]);
 
   useEffect(() => {
     if (

--- a/wormhole-connect/src/views/Bridge/NativeGasSlider.tsx
+++ b/wormhole-connect/src/views/Bridge/NativeGasSlider.tsx
@@ -120,7 +120,6 @@ function GasSlider(props: { disabled: boolean }) {
     const amountWithoutRelayerFee = Number.parseFloat(
       ((amount - (relayerFee || 0)) * 0.995).toFixed(8),
     );
-    console.log(amountWithoutRelayerFee);
     const actualMaxSwap =
       amount &&
       maxSwapAmt &&

--- a/wormhole-connect/src/views/Bridge/NativeGasSlider.tsx
+++ b/wormhole-connect/src/views/Bridge/NativeGasSlider.tsx
@@ -117,7 +117,10 @@ function GasSlider(props: { disabled: boolean }) {
     if (!amount || !maxSwapAmt || destGasPayment === PaymentOption.MANUAL)
       return;
     // multiply by 0.995 to avoid errors due to rounding, low amount, low gas, etc.
-    const amountWithoutRelayerFee = (amount - (relayerFee || 0)) * 0.995;
+    const amountWithoutRelayerFee = Number.parseFloat(
+      ((amount - (relayerFee || 0)) * 0.995).toFixed(8),
+    );
+    console.log(amountWithoutRelayerFee);
     const actualMaxSwap =
       amount &&
       maxSwapAmt &&

